state-loader.md

```diff
@@ -106,47 +106,23 @@ module STATE-LOADER
-    rule <k> load "account" : { ACCTID: { KEY : VALUE , REST } } => load "account" : { ACCTID : { KEY : VALUE } } ~> load "account" : { ACCTID : { REST } } ... </k> requires REST =/=K .JSONList
-
-    rule <k> load "account" : { ((ACCTID:String) => #parseAddr(ACCTID)) : ACCT }                                ... </k>
-    rule <k> load "account" : { (ACCT:Int) : { "balance" : ((VAL:String)         => #parseWord(VAL)) } }        ... </k>
-    rule <k> load "account" : { (ACCT:Int) : { "nonce"   : ((VAL:String)         => #parseWord(VAL)) } }        ... </k>
-    rule <k> load "account" : { (ACCT:Int) : { "code"    : ((CODE:String)        => #parseByteStack(CODE)) } }  ... </k>
-    rule <k> load "account" : { (ACCT:Int) : { "storage" : ({ STORAGE:JSONList } => #parseMap({ STORAGE })) } } ... </k>
-```
+    rule <k> load "pre" : { (ACCTID:String) : ACCT } => mkAcct #parseAddr(ACCTID) ~> loadAccount #parseAddr(ACCTID) ACCT ... </k>
 
-The individual fields of the accounts are dealt with here.
+    syntax EthereumCommand ::= "loadAccount" Int JSON
+ // -------------------------------------------------
+    rule <k> loadAccount ACCT { "balance" : (BAL:Int), REST } => loadAccount ACCT { REST } ... </k>
+         <account> <acctID> ACCT </acctID> <balance> _ => BAL </balance> ... </account>
 
-```{.k}
-    rule <k> load "account" : { ACCT : { "balance" : (BAL:Int) } } => . ... </k>
-         <account>
-           <acctID> ACCT </acctID>
-           <balance> _ => BAL </balance>
-           ...
-         </account>
-
-    rule <k> load "account" : { ACCT : { "code" : (CODE:ByteArray) } } => . ... </k>
-         <account>
-           <acctID> ACCT </acctID>
-           <code> _ => CODE </code>
-           ...
-         </account>
-
-    rule <k> load "account" : { ACCT : { "nonce" : (NONCE:Int) } } => . ... </k>
-         <account>
-           <acctID> ACCT </acctID>
-           <nonce> _ => NONCE </nonce>
-           ...
-         </account>
-
-    rule <k> load "account" : { ACCT : { "storage" : (STORAGE:Map) } } => . ... </k>
-         <account>
-           <acctID> ACCT </acctID>
-           <storage> _ => STORAGE </storage>
-           <origStorage> _ => STORAGE </origStorage>
-           ...
-         </account>
+    rule <k> loadAccount ACCT { "code" : (CODE:ByteArray), REST } => loadAccount ACCT { REST } ... </k>
+         <account> <acctID> ACCT </acctID> <code> _ => CODE </code> ... </account>
+
+    rule <k> loadAccount ACCT { "nonce" : (NONCE:Int), REST } => loadAccount ACCT { REST } ... </k>
+         <account> <acctID> ACCT </acctID> <nonce> _ => NONCE </nonce> ... </account>
+
+    rule <k> loadAccount ACCT { "storage" : (STORAGE:Map), REST } => loadAccount ACCT { REST } ... </k>
+         <account> <acctID> ACCT </acctID> <origStorage> _ => STORAGE </origStorage> <storage> _ => STORAGE </storage> ... </account>
+
+    rule <k> loadAccount _ { _ } => . ... </k> [owise]
```

Comment:
Maybe this should be turned into two rules (so that we handle it when junk keys are in the middle, but there are more keys afterwards that we do want to load):

```
    rule <k> loadAccount _ { .JSONList } => . ... </k>
    rule <k> loadAccount _ { _:String : _ , REST => REST } ... </k> [owise]
```